src/WebJobs.Extensions.CosmosDBCassandra/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -0,0 +1,143 @@
+        private static ISession session;
+
+
+        public CosmosDBTriggerListener(ITriggeredFunctionExecutor executor,
+            string functionId,
+            string keyspace,
+            string table,
+            bool startFromBeginning,
+            int feedpolldelay,
+            ICosmosDBCassandraService cosmosDBCassandraService,
+            ILogger logger)
+        {
+            this._logger = logger;
+            this._executor = executor;
+            this._functionId = functionId;
+            this._keyspace = keyspace;
+            this._table = table;
+            this._startFromBeginning = startFromBeginning;
+            this._defaultTimeSpan = feedpolldelay > 0 ? TimeSpan.FromMilliseconds(feedpolldelay) : TimeSpan.FromMilliseconds(5000);
+            this._cosmosDBCassandraService = cosmosDBCassandraService;
+            this._hostName = Guid.NewGuid().ToString();
+        }
+
+        public void Cancel()
+        {
+            this.StopAsync(CancellationToken.None).Wait();
+        }
+
+        public void Dispose()
+        {
+            session.Dispose();
+        }
+
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+
+            Cluster cluster = _cosmosDBCassandraService.GetCluster();
+
+            session = cluster.Connect(_keyspace);
+            //set initial start time for pulling the change feed
+
+            DateTime timeBegin = this._startFromBeginning ? DateTime.MinValue.ToUniversalTime() : DateTime.UtcNow;
+
+            //initialise variable to store the continuation token
+            byte[] pageState = null;
+            _logger.LogInformation(string.Format($"Reading from Cassandra API change feed..."));
+
+            while (!cancellationTokenSource.IsCancellationRequested)
+            {
+                try
+                {
+                    IStatement changeFeedQueryStatement = new SimpleStatement(
+                    $"SELECT * FROM " + _keyspace + "." + _table + $" where COSMOS_CHANGEFEED_START_TIME() = '{timeBegin.ToString("yyyy-MM-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture)}'");
+                    if (pageState != null)
+                    {
+                        changeFeedQueryStatement = changeFeedQueryStatement.SetPagingState(pageState);
+                    }
+                    RowSet rowSet = session.Execute(changeFeedQueryStatement);
+                    pageState = rowSet.PagingState;
+
+                    TimeSpan wait = _defaultTimeSpan;
+                    if (rowSet.IsFullyFetched)
+                    {
+                        List<Row> rowList = rowSet.ToList();
+                        CqlColumn[] columns = rowSet.Columns;
+                        if (rowList.Count != 0)
+                        {
+                            //convert Cassandra resultset to JArray
+                            List<JArray> rows = new List<JArray>();
```

Comment:
Why not do this conversion if the user is actually asking for JArray? Otherwise we are penalizing anyone that might ask for Row?

You can pack this converter as a function and use it in a:

```csharp
rule2.AddConverter<IReadOnlyList<Row>, IReadOnlyList<JArray>>(......);
```